app/controllers/v1/statuses_controller.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+module V1
+  # API for Compliance Status
+  class StatusesController < ApplicationController
+    skip_around_action :authenticate_user
+
+    ARB = ActiveRecord::Base
+
+    def show
+      render json: { data: all_statuses }, status: status_return_code
+    end
+
+    private
+
+    def all_statuses
+      @all_statuses ||= {
+        api: api_status
+      }
+    end
+
+    def status_return_code
+      all_statuses.values.all? ? :ok : :internal_server_error
+    end
+
+    def api_status
```

Comment:
What about something like?

```suggestion
    def api_status
      ARB.connection.active? || !!ARB.establish_connection
    rescue PG::Error
      false
    end
```

I have a feeling that you don't even need the `!!`

Reply:
I guess the `ARB.connection.active?` would be enough.

Reply:
I have never seen `ARB.establish_connection` ever return anything other than `false`. Are you seeing `true` when it reestablishes a connection?

`ARB.connection.active?` by itself doesn't seem to reestablish the connection once dropped.

Reply:
Does it have to return anything, considering Ruby's default return?
It might be a completely unrelated return from another function/method call

```
From: /home/vkrizan/.gem/gems/activerecord-import-1.0.4/lib/activerecord-import/import.rb:245 ActiveRecord::Import::Connection#establish_connection:

    243: def establish_connection(args = nil)
    244:   super(args)
 => 245:   ActiveRecord::Import.load_from_connection_pool connection_pool
    246: end
```
lets see what it does..

Reply:
This might be the one that the `false` comes from:
```
From: /home/vkrizan/.gem/gems/activerecord-import-1.0.4/lib/activerecord-import/base.rb:23 ActiveRecord::Import.require_adapter:

    22: def self.require_adapter(adapter)
 => 23:   require File.join(ADAPTER_PATH, "/#{base_adapter(adapter)}_adapter")
    24: rescue LoadError
    25:   # fallback
    26: end

[3] pry(ActiveRecord::Import)> require File.join(ADAPTER_PATH, "/#{base_adapter(adapter)}_adapter")
=> false
```